packages/flutter_migrate/lib/src/compute.dart

```diff
@@ -0,0 +1,1014 @@
+    targetFlutterDirectory,
+  );
+
+  // Create target reference app when not provided.
+  //
+  // This step directly calls flutter create with the target (the current installed revision)
+  // flutter sdk.
+  final MigrateTargetFlutterProject targetProject = MigrateTargetFlutterProject(
+    path: targetAppPath,
+    directory: targetProjectDir,
+    name: name,
+    androidLanguage: androidLanguage,
+    iosLanguage: iosLanguage,
+    platformWhitelist: platforms,
+  );
+  context.targetProject = targetProject;
+  await targetProject.createProject(
+    context,
+    revisionConfig.targetRevision,
+    targetFlutterDirectory,
+  );
+
+  await migrateUtils.gitInit(flutterProject.directory.absolute.path);
+
+  // Generate diffs. These diffs are used to determine if a file is newly added, needs merging,
+  // or deleted (rare). Only files with diffs between the base and target revisions need to be
+  // migrated. If files are unchanged between base and target, then there are no changes to merge.
+  status.pause();
+  logger.printStatus('Diffing base and target reference app.',
+      indent: 2, color: TerminalColor.grey);
+  status.resume();
+
+  context.migrateResult.diffMap
+      .addAll(await baseProject.diff(context, targetProject));
+
+  // Check for any new files that were added in the target reference app that did not
+  // exist in the base reference app.
+  status.pause();
+  logger.printStatus('Finding newly added files',
+      indent: 2, color: TerminalColor.grey);
+  status.resume();
+  context.migrateResult.addedFiles
+      .addAll(await baseProject.newlyAddedFiles(context, targetProject));
+
+  // Merge any base->target changed files with the version in the developer's project.
+  // Files that the developer left unchanged are fully updated to match the target reference.
+  // Files that the developer changed and were changed from base->target are merged.
+  status.pause();
+  logger.printStatus('Merging changes with existing project.',
+      indent: 2, color: TerminalColor.grey);
+  status.resume();
+  await MigrateFlutterProject.merge(
+    context,
+    baseProject,
+    targetProject,
+    unmanagedFiles,
+    unmanagedDirectories,
+    preferTwoWayMerge,
+  );
+
+  // Clean up any temp directories generated by this tool.
+  status.pause();
+  logger.printStatus('Cleaning up temp directories.',
+      indent: 2, color: TerminalColor.grey);
+  status.resume();
+  if (deleteTempDirectories) {
+    // Don't delete user-provided directories
+    if (!customBaseProjectDir) {
+      context.migrateResult.tempDirectories
+          .add(context.migrateResult.generatedBaseTemplateDirectory!);
+    }
+    if (!customTargetProjectDir) {
+      context.migrateResult.tempDirectories
+          .add(context.migrateResult.generatedTargetTemplateDirectory!);
+    }
+    context.migrateResult.tempDirectories
+        .addAll(context.migrateResult.sdkDirs.values);
+  }
+  status.stop();
+  return context.migrateResult;
+}
```

Comment:
This function is quite large, and doing a lot of things; extracting smaller, well-named helpers for individual parts would help with understanding the overall flow.